src/cursor.c

```diff
@@ -308,7 +314,7 @@ cursor_update_common(struct server *server, struct wlr_scene_node *node,
 		 * in response to the enter event.
 		 */
 		if (surface != wlr_seat->pointer_state.focused_surface
-				|| seat->cursor_set_by_server) {
+				|| seat->server_cursor) {
```

Comment:
I would make this `seat->server_cursor != LAB_CURSOR_CLIENT`; using the `enum` as a `bool` feels dirty.